src/webchat-ui/components/presentational/DisconnectOverlay.tsx

```diff
@@ -1,116 +1,109 @@
-	opacity: 0,
 	zIndex: 4,
 }));
 
-const Dialog = styled.div(({ theme }) => ({
-	padding: theme.unitSize * 2,
-	borderRadius: theme.unitSize,
-
-	textAlign: "center",
-}));
-
-const DialogHeader = styled.div(({ theme }) => ({
-	marginBottom: theme.unitSize * 2,
-	fontSize: 22,
-	fontWeight: 600,
-}));
-
 const HeaderIconButton = styled(IconButton)(({ theme }) => ({
 	position: "absolute",
 	right: 15,
 	top: 17,
-
-	color: theme.textLight,
-	fill: theme.textLight,
-	borderRadius: "50%",
+	borderRadius: 4,
+	"&:focus": {
+		outline: `2px solid ${theme.primaryColor}`,
+		outlineOffset: 2,
+	},
 }));
 
 interface DisconnectOverlayProps {
+	isOpen: boolean;
 	isPermanent: boolean;
 	onClose: () => void;
 	onConnect: () => void;
 	config: IWebchatConfig;
 }
 
-const DisconnectOverlay = ({ isPermanent, onClose, onConnect, config }: DisconnectOverlayProps) => {
+const DisconnectOverlay = (props: DisconnectOverlayProps) => {
+	const { isPermanent, onClose, onConnect, config, isOpen } = props;
+
+	const parentSelector = () => document.querySelector("#webchatWindow");
+
+	const closeRef = useRef<HTMLButtonElement>(null);
+
+	useEffect(() => {
+		// Hack around stolen autofocus after widget open
+		setTimeout(() => closeRef.current?.focus(), 200);
+	}, []);
+
 	return (
-		<Wrapper>
-			<Dialog>
-				{isPermanent ? (
-					<>
-						<DialogHeader>
-							{config.settings.customTranslations?.network_error ?? "Connection lost"}
-						</DialogHeader>
-						{navigator.onLine ? (
-							<Button onClick={onConnect} color="primary" style={{ margin: "auto" }}>
-								{config.settings.customTranslations?.reconnect ?? "Reconnect"}
-							</Button>
-						) : (
-							<div>
-								{config.settings.customTranslations?.no_network ??
-									"No network connection"}
-							</div>
-						)}
-					</>
-				) : (
-					<div>
-						<DialogHeader>
-							{config.settings.customTranslations?.network_error ?? "Connection lost"}
-						</DialogHeader>
-						<div>
-							{config.settings.customTranslations?.reconnecting ?? "Reconnecting..."}
-						</div>
-					</div>
-				)}
-			</Dialog>
+		<ReactModal
+			isOpen={true}
+			parentSelector={parentSelector}
+			onRequestClose={onClose}
+			style={{
+				overlay: {
+					position: "absolute",
+					zIndex: 3,
+					backgroundColor: "rgba(0, 0, 0, 0.5)",
+				},
+				content: {
+					inset: 50,
+					display: "flex",
+					flexDirection: "column",
+					alignItems: "center",
+					margin: "auto",
+					justifyContent: "center",
+				},
+			}}
+		>
 			<HeaderIconButton
+				autoFocus={!isPermanent} // Focus only if the reconnect button is not rendered
 				data-disconnect-overlay-close-button
 				onClick={onClose}
+				ref={closeRef}
 				className="webchat-header-close-button"
 				aria-label={
 					config.settings.customTranslations?.ariaLabels?.closeConnectionWarning ??
 					"Close connection lost overlay"
 				}
 			>
-				<CloseIcon />
+				<CloseIcon style={{ fill: "#000" }} />
```

Comment:
Would be nice to use a color from the theme

Reply:
fixed